### PR TITLE
Fix #43: add tests for client

### LINKED ISSUE #43: Add tests for client
I managed to merge #40 without realizing it was broken. There should be some tests for the client to avoid this situations.

### PR DESCRIPTION
**Pull request by circlespainter**
Should fix #43.

Letting the CI workflow also run client tests (and so build before that) should be enough to catch compile errors but I'm also proposing some client unit tests.

*Warning*: my first Rust code ahead so will most likely contain a fair amount of silliness.

### CODE REVIEW

**Comment by lightclient** [comment]
Hi @circlespainter! Thanks a lot for starting work on this :) I see you're still in draft mode, but I will start leaving a few comments so that you can iterate on them on your own schedule.

File: client/src/client/command.rs [hunk 1]
```diff
@@ -11,6 +11,8 @@ use std::collections::HashMap;
 
 /// A enum that describes the possible commands a user might send to the client and their required
 /// arguments.
+#[derive(Debug)]
+#[derive(PartialEq)]
```
Comment (lightclient) [inline client/src/client/command.rs]: You can actually combine these two: `#[derive(Debug, PartialEq)]`

**Comment by circlespainter** [comment]
> Hi @circlespainter! Thanks a lot for starting work on this :) I see you're still in draft mode, but I will start leaving a few comments so that you can iterate on them on your own schedule.

Absoutely, thanks!

**Comment by lightclient** [comment]
Hi @circlespainter, sorry for the delay in responding. I have been AFK on vacation for a couple weeks! Thanks a lot for doing this, everything looks good.

> Adding a test for the OK case of Transfer (by using a mock HTTP server) seems to be non-trivial
You're totally right. I think it would be easier if things were a little more abstracted, but for now the HTTP call is just right inside that command. I wouldn't worry about it too much, I think what you've done here is more than enough for the current stage of the project.

> imp::value seem to panic rather than returning an error if the address doesn't exist
Yeah, that does seem like a bug! If you want to go ahead with opening that issue, that would be awesome.

Thanks again!

**Pull request merged**